frontend/src/components/pages/ChoresPage.js

```diff
@@ -173,53 +190,80 @@ const ChoresPage = () => {
+                    d="M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5v11z"
+                  />
                 </svg>
               </button>
-            )
+            );
           } else {
             return "";
           }
         },
       },
       {
         id: "delete",
-        accessor: data => data.id,
-        Cell: ({value}) => {
-          if(value !== -1) {
+        accessor: (data) => data.id,
+        Cell: ({ value }) => {
+          if (value !== -1 && !isChoresLoading) {
             return (
-              <button className="deleteButton" onClick={() => {
-                let chore = chores.find((val) => {return value === val._id});
-                setCurrChore(chore);
-                handleDeleteModalShow();
-              }}>
-                <svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" fill="currentColor" className="bi bi-trash3-fill" viewBox="0 0 16 16">
-                  <path d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5Zm-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5ZM4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06Zm6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528ZM8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5Z"/>
+              <button
+                className="deleteButton"
+                onClick={() => {
+                  let chore = choresData.data.find((val) => {
+                    return value === val._id;
+                  });
+                  setCurrChore(chore);
+                  handleDeleteModalShow();
+                }}
+              >
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
+                  width="1em"
+                  height="1em"
+                  fill="currentColor"
+                  className="bi bi-trash3-fill"
+                  viewBox="0 0 16 16"
+                >
+                  <path d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5Zm-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5ZM4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06Zm6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528ZM8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5Z" />
                 </svg>
               </button>
-            )
+            );
           } else {
             return "";
           }
         },
-      }
+      },
     ],
-    [chores]
+    [choresData]
   );
 
   // Chore data and table setup
-  const data = useMemo(() => getChoreData(chores), [chores]);
+  const data = useMemo(() => formatChoreData(), [choresData]);
```

Comment:
Also a warning generated here to add formatChoreData to the dependency array. I think you might just need to pass choresData into formatChoreData

Reply:
We no longer need to pass choresData into formatChoreData as it just accesses it directly, but I did forget to remove the arrow function around it, that way it no longer tells us to include formatChoreData in the dependency array. It did require adding choresData and isChoresLoading to the dependency array since those are used in the function, so I did that and all is well.